### PR TITLE
Add requirements file loading to install-opkg and download-opkg

### PR DESCRIPTION
**Pull request by CrispyBacon1999**
Load in the opkgs from a requirements.txt formatted file. Supports comments with lines starting with # and semicolon delimited packages. Should match the requirements.txt [formatting spec](https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format) pretty closely.

### CODE REVIEW

**Comment by auscompgeek** [comment]
Why semicolon delimited? Not sure why that'd be useful in a file listing packages.

**Comment by CrispyBacon1999** [comment]
I apparently misinterpreted what semicolons are used for in the regular requirements.txt file.

File: robotpy_installer/installer.py [hunk 1]
```diff
@@ -1076,13 +1076,39 @@ def download_opkg_opts(self, parser):
...
+    def _load_opkg_from_req(*files):
+        """
+            Pull the list of opkgs from the files
+        """
+        opkgs = []
+        for file in files:
+            with open(file, "r") as f:
+                for row in f.readlines():
+                    if not row.startswith("#"):
+                        opkgs.append(row)
```
Comment (virtuald) [inline robotpy_installer/installer.py]: What about trailing whitespace in the file? Also, is there a way to specify versions?

Comment (CrispyBacon1999) [reply robotpy_installer/installer.py]: Got this fixed. Version control doesn't seem to be a thing that's built into the installer currently or at least it isn't documented.

**Comment by virtuald** [comment]
You added a bunch of files in that last commit, can you remove them? Otherwise this looks good to go for me.

**Comment by CrispyBacon1999** [comment]
Should be all set to go now, completely forgot about the opkg_cache folder being made when i was testing it.

File: robotpy_installer/win32/plink.exe.jmd [hunk 2]
```diff
@@ -0,0 +1 @@
+{"etag": "\"92ba8-593211b5bdccd\"", "last-modified": "Sun, 22 Sep 2019 09:41:25 GMT", "md5": "94ea5d0f573e93bcfc710a7c1a346c75"}
```
Comment (auscompgeek) [inline robotpy_installer/win32/plink.exe.jmd]: ?

**Comment by auscompgeek** [comment]
Note to self: squash merge this to remove the bogus files from history.

File: robotpy_installer/installer.py [hunk 3]
```diff
@@ -1084,6 +1091,21 @@ def install_opkg_opts(self, parser):
...
+        """
+            Pull the list of opkgs from the files
+        """
+        opkgs = []
+        # Loop through the passed in files to support multiple requirements files
+        for file in files:
+            with open(file, "r") as f:
+                for row in f.readlines():
+                    # Ignore commented lines and empty lines
+                    if row.strip() and not row.startswith("#"):
```
Comment (auscompgeek) [inline robotpy_installer/installer.py]: Why not strip once before the if statement instead of stripping twice?

Comment (CrispyBacon1999) [reply robotpy_installer/installer.py]: That would probably make more sense. Original thought was to preserve file formatting but also could be slightly bugged if there's a space before the #. And also the formatting of the file doesn't really matter when it's being parsed.

**Comment by CrispyBacon1999** [comment]
> 
> 
> Why have the file modes changed?

?

**Comment by virtuald** [comment]
Thanks for the changes.

**Pull request merged**